src/daft-sql/src/planner.rs

```diff
@@ -55,49 +84,78 @@ impl Relation {
             alias: None,
         }
     }
+
     pub fn with_alias(self, alias: TableAlias) -> Self {
         Self {
             alias: Some(alias),
             ..self
         }
     }
+
     pub fn get_name(&self) -> String {
         self.alias
             .as_ref()
             .map(|alias| ident_to_str(&alias.name))
             .unwrap_or_else(|| self.name.clone())
     }
+
     pub(crate) fn schema(&self) -> SchemaRef {
         self.inner.schema()
     }
 }
 
-/// Context that is shared across a query and its subqueries
+/// Context for the planning the statement.
+/// TODO consolidate SQLPlanner state to the single context.
+/// TODO move bound_ctes into per-planner scope since these are a scoped concept.
 #[derive(Default)]
 struct PlannerContext {
-    catalog: DaftCatalog,
-    cte_map: HashMap<String, Relation>,
+    /// Session provides access to metadata and the path for name resolution.
+    /// TODO move into SQLPlanner once state is flipped.
+    /// TODO consider decoupling session from planner via a resolver trait.
+    session: Arc<Session>,
```

Comment:
slightly unrelated: while we're in here if it's not too much extra work, we should change the PlannerContext to take in a lifetime so we can borrow the session. 
We don't ever actually need an owned session or catalog as we just need read access to check for the existence of tables. 

```rs
struct PlannerContext<'session> {
  session: &'session Session
}

impl PlannerContext<'_> 
  fn new(session: &Session) -> Self { }
```

Reply:
Totally agreed, was somewhat curious too why daft generally uses arcs over refs+lifetimes. I like the ref+lifetime here.

Reply:
I think its mostly for simplicity. A lot of our code isn't performance sensitive, so cloning arcs is generally fine, but it's generally not considered best practice.

Reply:
I may have to update this is a later PR when I refactor how the planner manages state. The lifetime ended up polluting a lot of methods, and the nested planners made this worse vs putting them all on the same lifetime. For the time being, I think Rc is simple enough. Let me know if you think that's fine, otherwise I can try to finish introducing the reference.